habitat_baselines/common/base_trainer.py

```diff
@@ -98,11 +98,19 @@ def eval(self) -> None:
             ), "Must specify a directory for storing videos on disk"
 
         with get_writer(self.config, flush_secs=self.flush_secs) as writer:
-            if os.path.isfile(self.config.EVAL_CKPT_PATH_DIR):
+            if (
+                os.path.isfile(self.config.EVAL_CKPT_PATH_DIR)
+                or not self.config.EVAL.SHOULD_LOAD_CKPT
```

Comment:
Why does SHOULD_LOAD_CKPT influence wether or not we should evaluate a single checkpoint? Is I have no idea what SHOULD_LOAD_CKPT does at all but it looks like if 
```
SHOULD_LOAD_CKPT = False
EVAL_CKPT_PATH_DIR = directory
```
Then you will call _eval_checkpoint with a directory which will raise an error.

Why is this change needed ?

Reply:
I think it is best this way actually. When `SHOULD_LOAD_CKPT = False` only a single eval should run no matter what since `EVAL_CKPT_PATH_DIR`. So the situation you described is okay since `EVAL_CKPT_PATH_DIR` is ignored. 

This change is necessary because for models that do not load from a single checkpoint we should not iterate through the default `EVAL_CKPT_PATH_DIR` argument (which is a directory by default).

Reply:
I added a comment about this.